client-admin/webpack.config.js

```diff
@@ -2,61 +2,155 @@
+  var isDevBuild = options.mode === 'development';
+  var isDevServer = process.env.WEBPACK_SERVE;
+  var chunkHashFragment = (isDevBuild || isDevServer) ? '' : '.[chunkhash:8]';
+  return {
+    entry: ["./src/index"],
+    output: {
+      publicPath: '/',
+      filename: `static/js/admin_bundle${chunkHashFragment}.js`,
+      path: path.resolve(__dirname, "dist"),
+      clean: true,
+    },
+    resolve: {
+      extensions: [".js", ".css", ".png", ".svg"],
+    },
+    devServer: {
+      historyApiFallback: true,
+      /**
+      // TODO: Set up API proxy later for server component.
+      // See: https://webpack.js.org/configuration/dev-server/#devserverproxy
+      proxy: {
+        '/api': {
+        target: 'https://pol.is',
+        secure: false,
         },
       },
-      {
-        test: /\.(png|jpg|gif|svg)$/,
-        loader: "file-loader",
-      },
-      {
-        test: /\.mdx?$/,
-        use: ['babel-loader', '@mdx-js/loader']
-      }
+      **/
+    },
+    plugins: [
+      new CopyPlugin({
+        patterns: [
+          { from: 'public', globOptions: { ignore: ['**/index.html']}},
+        ],
+      }),
+      new HtmlWebPackPlugin({
+        template: path.resolve( __dirname, 'public/index.html' ),
+        filename: (isDevBuild || isDevServer) ? 'index.html' : 'index_admin.html',
+        templateParameters: {
+          domainWhitelist: `["${polisConfig.domainWhitelist.join('","')}"]`,
+          fbAppId: polisConfig.FB_APP_ID,
+          usePlans: !isTrue(polisConfig.DISABLE_PLANS),
+          useIntercom: !isTrue(polisConfig.DISABLE_INTERCOM),
+        },
+      }),
+      new LodashModuleReplacementPlugin({
+        currying: true,
+        flattening: true,
+        paths: true,
+        placeholders: true,
+        shorthands: true
+      }),
+      // Only run analyzer when specified in flag.
+      ...(cliArgs.analyze ? [new BundleAnalyzerPlugin({ defaultSizes: 'gzip' })] : []),
+      // Only compress and create headerJson files during production builds.
+      ...((isDevBuild || isDevServer || cliArgs.analyze) ? [] : [
+        new CompressionPlugin({
+          test: /\.js$/,
+          // Leave unmodified without gz ext.
+          // See: https://webpack.js.org/plugins/compression-webpack-plugin/#options
+          filename: '[path][base]',
+          deleteOriginalAssets: true,
+        }),
+        new EventHooksPlugin({
+          afterEmit: () => {
+            console.log('Writing *.headersJson files...')
+
+            function writeHeadersJson(matchGlob, headersData = {}) {
+              const files = glob.sync(path.resolve(__dirname, "dist", matchGlob))
+              files.forEach((f, i) => {
+                const headersFilePath = f + '.headersJson'
+                fs.writeFileSync(headersFilePath, JSON.stringify(headersData))
+              })
+            }
+
+            function writeHeadersJsonHtml() {
+              const headersData = {
+                'x-amz-acl': 'public-read',
```

Comment:
The 'x-amz'acl' header is S3 specific, and the whole .headersJson seems to be an S3 thing. Since you removed this from the codebase do we need it still?

Reply:
Hi @brendanarnold. I think we still want the `.headersJson` and S3 data because we deploy assets to S3+Cloudflare CDN. But thanks for laying some scrutiny on this!

Reply:
Just saw that I never responded.

What's happening here is that for each file that is getting spit out, an additional `.headersJson` file is being created, which is then used in a separate process when we upload to S3 (which can be set to automatically attach this content as metadata in responses, so that caching and content-type come out correctly). If this doesn't seem like the best way for things to work, that's probably cause you're right. This is very old code base stuff, some of which was coded around constraints that don't exist anymore.

Reply:
heh I was always curious exactly how the headersJson stuff worked, and what portions of this were still actively used :)